packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -845,6 +846,7 @@ export class KubernetesVersion {
 
   /**
    * Kubernetes version 1.21
+   * @deprecated Use newer version of EKS
```

Comment:
I would prefer we not deprecate `1.21`, at least not in this PR. It is the legacy default for a lot of things and I think we want to think more about what this entails first (and maybe have it explicitly called out in the release notes as its own PR).

Reply:
Reason why I added deprecation notice was because 1.21 support ended on Feb 15th, 2023 as per https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html#kubernetes-release-calendar. I can still roll this back if you want me to. Just LMK.

Reply:
I would prefer rollback and put it in a new PR. At the very least we should call out the deprecation in the release notes, and I want to give others the opportunity to refute the deprecation regardless of rolling out 1.25

Reply:
Removed for now. :)